src/spans/awsSpan.ts

```diff
@@ -66,6 +68,79 @@ export const getSpanInfo = (): SpanInfo => {
   return { traceId, tracer, logGroupName, logStreamName };
 };
 
+export const getSpanPriority = (span): number => {
+  if (span.type === FUNCTION_SPAN) {
+    return 0;
+  }
+  if (span.type === ENRICHMENT_SPAN) {
+    return 1;
+  }
+  if (spanHasErrors(span)) {
+    return 2;
+  }
+  return 3;
+};
+
+export const spansPrioritySorter = (span1: any, span2: any): number => {
+  const span1Priority = getSpanPriority(span1);
+  const span2Priority = getSpanPriority(span2);
+
+  if (span1Priority < span2Priority) {
+    return -1;
+  } else if (span1Priority > span2Priority) {
+    return 1;
+  } else {
+    return 0;
+  }
+};
+
+export const getSpanMetadata = (span: any): any => {
+  const spanCopy = JSON.parse(JSON.stringify(span));
```

Comment:
So the benchmark is cool with that? :)

Reply:
yes, So I think we are good. (We do this logic only in rare cases so most of the time it will not run at all